test/parallel/test-zlib-dictionary.js

```diff
@@ -69,6 +69,66 @@ function run(num) {
 }
 run(1);
 
+function rawDictionaryTest() {
+  let output = '';
+  const deflate = zlib.createDeflateRaw({ dictionary: spdyDict });
+  const inflate = zlib.createInflateRaw({ dictionary: spdyDict });
+
+  deflate.on('data', function(chunk) {
+    inflate.write(chunk);
+  });
+
+  inflate.on('data', function(chunk) {
+    output += chunk;
+  });
+
+  deflate.on('end', function() {
+    inflate.end();
+  });
+
+  inflate.on('end', common.mustCall(function() {
+    assert.equal(input, output);
```

Comment:
So just to be clear, do you want me to backport both of the commits from last time, both the actual fix and the test fixes? I thought you said just to backport the first, that's why I didn't include the test refactors, but I might have misunderstood. I'd be happy to do them both if that's what you want.